stdlib/SuiteSparse/src/spqr.jl

```diff
@@ -419,7 +424,8 @@ function _ldiv_basic(F::QRSparse, B::StridedVecOrMat)
                         view(X0, Base.OneTo(rnk), :))
 
     # Apply right permutation and extract solution from X
-    return getindex(X, ntuple(i -> i == 1 ? invperm(F.cpiv) : :, Val(ndims(B)))...)
+    # NB: cpiv == [] if SPQR was called with ORDERING_FIXED
+    return length(F.cpiv) == 0 ? getindex(X, ntuple(i -> i == 1 ? (1:size(F,2)) : :, Val(ndims(B)))...) : getindex(X, ntuple(i -> i == 1 ? invperm(F.cpiv) : :, Val(ndims(B)))...)
```

Comment:
Style nit: please keep line length reasonable. So usually about 1 anonymous function per expression.

Reply:
I had trouble with this when I tested. That's what I try to explain above. Adding a return statement for the special case causes threaded tests to fail. I'm not sure why. I'll try again.

Reply:
Oh well, nevermind. I cleaned it up and it seems ok.